db/listener_test.cc

```diff
@@ -515,6 +515,9 @@ TEST_F(EventListenerTest, DisableBGCompaction) {
     db_->GetColumnFamilyMetaData(handles_[1], &cf_meta);
   }
   ASSERT_GE(listener->slowdown_count, kSlowdownTrigger * 9);
+  // We don't want the listener executing during DBTestBase::Close() due to
+  // race on handles_.
+  dbfull()->TEST_WaitForBackgroundWork();
```

Comment:
How about also asserting on `error_handler_.GetBGError()` is OK?

Reply:
Better, I'll make `TEST_WaitForBackgroundWork()` return `error_handler_.GetBGError()` (like other similar functions) and `ASSERT_OK` on that.